pcsx2/gui/Dialogs/GSDumpDialog.cpp

```diff
@@ -0,0 +1,800 @@
+}
+
+void Dialogs::GSDumpDialog::GenPacketList(std::vector<GSData>& dump)
+{
+	int i = 0;
+	m_gif_list->DeleteAllItems();
+	wxTreeItemId mainrootId = m_gif_list->AddRoot("root");
+	wxTreeItemId rootId = m_gif_list->AppendItem(mainrootId, "0 - VSync");
+	for (auto& element : dump)
+	{
+		wxString s;
+		([&] { return element.id == Transfer; })() ? (s.Printf("%d - %s - %s - %d byte", i, GSTypeNames[element.id], GSTransferPathNames[element.path], element.length)) : 
+									s.Printf("%d - %s - %d byte", i, GSTypeNames[element.id], element.length);
+		if (element.id == VSync)
+		{
+			m_gif_list->SetItemText(rootId, s);
+			rootId = m_gif_list->AppendItem(mainrootId, "VSync");
+		}
+		else 
+			m_gif_list->AppendItem(rootId, s);
+		i++;
+	}
+	m_gif_list->Delete(rootId);
+}
+
+void Dialogs::GSDumpDialog::GenPacketInfo(GSData& dump)
+{
+	m_gif_packet->DeleteAllItems();
+	wxTreeItemId rootId = m_gif_packet->AddRoot("root");
+	switch (dump.id)
+	{
+			case Transfer:
+			{
+				wxTreeItemId trootId;
+				wxString s;
+				s.Printf("Transfer Path %s", GSTransferPathNames[dump.path]);
+				trootId = m_gif_packet->AppendItem(rootId, s);
+				u64 tag = *(u64*)(dump.data);
+				u64 regs = *(u64*)(dump.data + 8);
+				u8 nloop = tag & ((u64)(1 << 15) - 1);
+				u8 eop = (tag >> 15) & 1;
+				u8 pre = (tag >> 46) & 1;
+				u32 prim = (tag >> 47) & ((u64)(1 << 11) - 1);
+				u8 flg = ((tag >> 58) & 3);
+				u32 nreg = (u32)((tag >> 60) & ((u64)(1 << 4) - 1));
+				if (nreg == 0)
+					nreg = 16;
+
+				std::vector<wxString> infos(7);
+				m_stored_q = 1.0;
+
+				infos[0].Printf("nloop = %u", nloop);
+				infos[1].Printf("eop = %u", eop);
+				infos[2].Printf("flg = %s", GifFlagNames[flg]);
+				infos[3].Printf("pre = %u", pre);
+				infos[4].Printf("Prim");
+				infos[5].Printf("nreg = %u", nreg);
+				infos[6].Printf("reg");
+
+				wxTreeItemId primId;
+				wxTreeItemId regId;
+				for (int i = 0; i < 7; i++)
+				{
+					wxTreeItemId res = m_gif_packet->AppendItem(trootId, infos[i]);
+					switch (i)
+					{
+						case 4:
+							ParseTreePrim(res, prim);
+							break;
+						case 6:
+							regId = res;
+							break;
+					}
+				}
+
+				int p = 16;
+				switch ((GifFlag)flg)
+				{
+					case GIF_FLG_PACKED:
+					{
+						for (int j = 0; j < nloop; j++)
+						{
+							for (u32 i = 0; i < nreg; i++)
+							{
+								u128 reg_data;
+								reg_data.lo =  *(u64*)(dump.data + p);
+								reg_data.hi =  *(u64*)(dump.data + p + 8);
+								ParseTreeReg(regId, (GIFReg)((regs >> (i * 4)) & ((u64)(1 << 4) - 1)), reg_data, true);
+								p += 16;
+							}
+						}
+						break;
+					}
+					case GIF_FLG_REGLIST:
+					{
+						for (int j = 0; j < nloop; j++)
+						{
+							for (u32 i = 0; i < nreg; i++)
+							{
+								u128 reg_data;
+								reg_data.lo =  *(u64*)(dump.data + p);
+								ParseTreeReg(regId, (GIFReg)((regs >> (i * 4)) & ((u64)(1 << 4) - 1)), reg_data, false);
+								p += 8;
+							}
+						}
+						break;
+					}
```

Comment:
Braces not needed for these cases

Reply:
Actually I'm curious about your thoughts regarding all this, if a case requires a variable assignment, shouldn't we make all cases use brackets for readability sake? That's why I kept using brackets within the whole file

Reply:
went on and removed unneeded braces, if someone complains i'll put them back